### PR TITLE
Create a destop application with Electron

### PR DESCRIPTION
**Pull request by leosuncin**
Use Electron to package the web application, so it can work offline.
Still I didn't implement a way to package and share the app.

Please review it and will continue improve it.

### CODE REVIEW

**Comment by leosuncin** [comment]
Run with ``npm run start:electron```

File: package.json [hunk 1]
```diff
@@ -2,9 +2,11 @@
   "name": "typist",
   "version": "0.5.0",
   "private": true,
+  "main": "public/electron.js",
```
Comment (vimark1) [inline package.json]: Is this needed?

File: public/electron.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const electron = require('electron');
```
Comment (vimark1) [inline public/electron.js]: can this file sit in the root of the project?

File: public/electron.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+app.on('window-all-closed', () => {
+  if (process.platform !== 'darwin') {
+    app.quit();
+  }
+});
+
+app.on('activate', () => {
+  if (mainWindow === null) {
+createWindow();
```
Comment (vimark1) [inline public/electron.js]: indentation issue

**Comment by vimark1** [comment]
Hey @leosuncin - it's looking good.   I've reviewed and a left some minor comments.
Just have 2 other general comments:
1. The login isn't working, we use Firebase auth so that may not be possible to work with electron?
2. Is it possible to include a build step to package the app for distribution?

File: package.json (hunk 1 above)
Comment (leosuncin) [reply package.json]: Electron read the `package.json` and look for `main` value, is the entrypoint for Electron

File: public/electron.js (hunk 2 above)
Comment (leosuncin) [reply public/electron.js]: When run ``npm run build`` the content of `public` is copy to `build` on production

File: public/electron.js (hunk 3 above)
Comment (leosuncin) [reply public/electron.js]: Fix at 013f7569866fe434944eb22dad91216846657702

**Comment by leosuncin** [comment]
1. I'll fix Firebase authentication with Electron, due it works different on electron than the browser.
2. After you reviewed the progress I was planned to use `electron-build` to package the app.

**Comment by vimark1** [comment]
Closing as this is fairly old now

**Pull request closed without merging**